### PR TITLE
Validate mainnet in CI

### PR DESCRIPTION
**Pull request by nc6**
This adds a test to convert and validate the mainnet mirror during CI.

It additionally fixes a bug which is currently stopping mainnet from validating.

We only do the immutable DB validation, rather than full ledger validation, since ledger validation takes a few hours for all of mainnet.

### CODE REVIEW

File: nix/validate-mainnet.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{ pkgs
+, byron-db-converter
+}:
+
+let
+  cardano-mainnet-config = pkgs.fetchurl {
+    url = https://raw.githubusercontent.com/input-output-hk/cardano-node/master/mainnet-genesis.json;
```
Comment (rvl) [inline nix/validate-mainnet.nix]:
```suggestion
    url = https://raw.githubusercontent.com/input-output-hk/cardano-node/114ee7f3b1cb55d384f928552c6b0871d4ca27ff/configuration/mainnet-genesis.json;
```

File: ouroboros-consensus/tools/db-convert/Main.hs [hunk 2]
```diff
@@ -159,14 +162,22 @@ validateChainDb
...
   -> IO ()
-validateChainDb dbDir cfg verbose =
-  ResourceRegistry.with $ \registry -> do
-    chaindb <- give protocolMagicId $ give epochSlots $ ChainDB.openDB $ args registry
-    blk <- ChainDB.getTipBlock chaindb
-    putStrLn $ "DB tip: " ++ condense blk
-    ChainDB.closeDB chaindb
+validateChainDb dbDir cfg onlyImmDB verbose =
+  ResourceRegistry.with $ \registry -> if onlyImmDB
+    then give protocolMagicId $ give epochSlots $ do
```
Comment (mrBliss) [inline ouroboros-consensus/tools/db-convert/Main.hs]: Maybe the two `give`s can be moved up a level so there's no duplication?

File: ouroboros-consensus/tools/db-convert/Main.hs [hunk 3]
```diff
@@ -159,14 +162,22 @@ validateChainDb
...
+      let (immDBArgs, _, _, _) = Args.fromChainDbArgs $ args registry
+      immDB <- ImmDB.openDB immDBArgs
+      immDbTipBlock <- ImmDB.getBlockAtTip immDB
+      putStrLn $ "DB tip: " ++ condense immDbTipBlock
+      ImmDB.closeDB immDB
```
Comment (mrBliss) [inline ouroboros-consensus/tools/db-convert/Main.hs]: Best to use `bracket` for the open and close (also in the `else`)

**Pull request merged**